Source/MQTTnet/Server/MqttSubscriptionInterceptorContext.cs

```diff
@@ -20,8 +25,41 @@ public MqttSubscriptionInterceptorContext(string clientId, TopicFilter topicFilt
         /// </summary>
         public IDictionary<object, object> SessionItems { get; }
 
-        public bool AcceptSubscription { get; set; } = true;
+        public bool AcceptSubscription
+        {
+            get => _acceptSubscription;
+            set {
+                if (!value && _resultCode < MqttSubscribeReasonCode.UnspecifiedError)
```

Comment:
Is this really required? I want to avoid implicit code in setters of properties.

Reply:
There i a problem that a new variable has been introduced and you now need to keep the result code error and the accept subscription state synchronized.  Keeping the backwards compatibility on AcceptSubscription means that old code which just sets that value needs to modify the returned error code as well.  I agree that is is not ideal, but I am not sure how else to address it.

Reply:
Ok got your point. You mean that "AcceptSubscription" cannot be true and Reason code something like "denied" at the same time right? I am not 100 % sure right now but I assume we have to declare the "AcceptSubscription" as obsolete and only use the result code. Then no special setter code is required. But for backward compatibility we have to keep the accept subscription for a small amount of versions. Maybe we can check the "AcceptSubscription" for true only and return a proper "best matching" result code. Or we can set the "AcceptSubscription" to a nullable boolean!? What do you think?

Reply:
I had been thinking about moving to a single variable as well.  I did keep the setter on AcceptSubscription.  I think that AcceptSubscription should get the getter and lose the setter in the future.  I tried to just obsolete that one item and was told that it needed to move to a language version of 8.2 in order for the obsolete attribute to be placed on just the setter.  I was reluctant to obsolete the entire attribute as keeping the error checking code in one location seems to be a better answer as far as I was concerned.